### PR TITLE
feat(): ECS service and checks

### PR DESCRIPTION
**Pull request by n4ch04**
### Context 

It is needed to migrate ECS and its checks from version 2 to 3

### Description

ECS service and checks

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: providers/aws/services/ecs/ecs_service.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        threads = []
+        for regional_client in self.regional_clients.values():
+            threads.append(threading.Thread(target=call, args=(regional_client,)))
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+    def __list_task_definitions__(self, regional_client):
+        logger.info("ECS listing task definitions...")
```
Comment (jfagoagas) [inline providers/aws/services/ecs/ecs_service.py]:
```suggestion
        logger.info("ECS - Listing Task Definitions...")
```

File: providers/aws/services/ecs/ecs_service.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+                            region=regional_client.region,
+                            environment_variables=[],
+                        )
+                    )
+        except Exception as error:
+            logger.error(
+                f"{regional_client.region} -- {error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def __describe_task_definition__(self):
```
Comment (jfagoagas) [inline providers/aws/services/ecs/ecs_service.py:53]:
```suggestion
    def __describe_task_definition__(self):
    logger.info("ECS - Describing Task Definitions...")
        try:
```

Comment (jfagoagas) [reply providers/aws/services/ecs/ecs_service.py:53]: Missing try-except clause

File: providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.metadata.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
+{
+    "Provider": "aws",
+    "CheckID": "ecs_task_definitions_no_environment_secrets",
+    "CheckTitle": "Check if there exists secrets in ECS task definitions environment variables",
```
Comment (jfagoagas) [inline providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.metadata.json]:
```suggestion
    "CheckTitle": "Check if secrets exists in ECS task definitions environment variables",
```

File: providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.metadata.json [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    "CheckType": ["Protect", "Secure development", "Credentials not hard-coded"],
+    "ServiceName": "ecs",
+    "SubServiceName": "",
+    "ResourceIdTemplate": "arn:partition:service:region:account-id:resource-id",
+    "Severity": "critical",
+    "ResourceType": "AwsEcsTaskDefinition",
+    "Description": "Check if there exists secrets in ECS task definitions environment variables",
```
Comment (jfagoagas) [inline providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.metadata.json]:
```suggestion
    "Description": "Check if secrets exists in ECS task definitions environment variables",
```

File: providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.metadata.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "Risk": "The use of a hard-coded password increases the possibility of password guessing. If hard-coded passwords are used; it is possible that malicious users gain access through the account in question.",
+    "RelatedUrl": "",
+    "Remediation": {
+      "Code": {
+        "CLI": "aws ecs describe-task-definition --region <REGION> --task-definition <TASK_DEFINITION_NAME> --query taskDefinition.containerDefinitions[*].environment",
```
Comment (jfagoagas) [inline providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.metadata.json]:
```suggestion
        "CLI": "",
```

File: providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+                for env_var in task_definition.environment_variables:
+                    dump_env_vars = {}
+                    dump_env_vars.update({env_var.name: env_var.value})
+
+                temp_env_data_file = tempfile.NamedTemporaryFile(delete=False)
+
+                env_data = b64decode(
+                    b64encode(dumps(dump_env_vars).encode("utf-8"))
+                ).decode("utf-8")
+                temp_env_data_file.write(bytes(env_data, encoding="raw_unicode_escape"))
```
Comment (jfagoagas) [inline providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.py]: Check this, you don't need the `b64` operations, use the `dump_env_vars` values as plain text.

File: providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets_test.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+AWS_REGION = "eu-west-1"
+AWS_ACCOUNT_NUMBER = "123456789012"
+task_name = "test-task"
+env_var_name_no_secrets = "host"
+env_var_value_no_secrets = "localhost:1234"
+env_var_name_with_secrets = "DB_PASSWORD"
+env_var_value_with_secrets = "pass-12343"
+
+
+class Test_ecs_task_definitions_no_environment_secrets:
```
Comment (jfagoagas) [inline providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets_test.py:15]: I would include a test case checking no task definitions.

**Review by jfagoagas — DISMISSED** [review]
Good job, please check my comments to merge.

File: providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets_test.py (hunk 7 above)
Comment (MrCloudSec) [reply providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets_test.py:15]: Done

File: providers/aws/services/ecs/ecs_service.py (hunk 2 above)
Comment (MrCloudSec) [reply providers/aws/services/ecs/ecs_service.py:53]: Done

File: providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.py (hunk 6 above)
Comment (MrCloudSec) [reply providers/aws/services/ecs/ecs_task_definitions_no_environment_secrets/ecs_task_definitions_no_environment_secrets.py]: Done

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**